### ISSUE TITLE
Display ping info for each server in the multiplayer menu

### ISSUE THREAD

**Issue by rvstwi** [OpenRVS-devs/OpenRVS#8]
Servers should display a reasonably accurate ping in the list of servers.

I tried doing this quite a while ago but couldn't get it to work.  If I recall there were issues updating each menu item in the server list menu - couldn't seem to get it to update?

Shouldn't be too hard to measure latency when sending the query to individual servers.

**Comment by ijemafe** [comment]
Will try this today

**Comment by ijemafe** [comment]
I was able to get server latency working in-game by adding the following to `QueryForServerInfo()`:
```
addr = Left(CurServer.szIPAddr, InStr(CurServer.szIPAddr, ":"));
CurServer.iPing = m_GameService.NativeGetPingTime(addr);
```
However, this ping can take up to 1000ms and they all happen sequentially. I tried moving the call to its own class to see if that would parallelize it, but the game hangs for 10 seconds when you open MP or click refresh.

Is there a good way to parallelize the `NativeGetPingTime()` calls?

**Comment by rvstwi** [comment]
Hmm great question.  I've played with that function in the past as well but can't remember why I abandoned working on it.  I'd imagine that it would be easier to keep track of when the client sends the query to each server in uscript, and then compare against when the response is received.  Rather than trying to figure out what Ubisoft's native code did.  Could be wrong though.

**Comment by rvstwi** [comment]
Branch ping-dev has our work on this so far.  It's working very nicely but we need to fix up how refresh and sorting works so ping (and other details) don't just disappear when we try to sort by latency or player count or something.  After that's fixed I'll merge with master.

**Comment by rvstwi** [comment]
This issue can be closed by push #27 .  Note that sorting is still an issue, but that's related to something else so I'll open a new issue for it.
